File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/per5java/Farmacia.java
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
public class Farmacia implements Serializable {
    private String nombreFarmacia;
    private int sucursal;
    private String direccion;
    private List<Medicamento> medicamentos;
    // Constructor
    public Farmacia(String nombreFarmacia, int sucursal, String direccion) {
        this.nombreFarmacia = nombreFarmacia;
        this.sucursal = sucursal;
        this.direccion = direccion;
        this.medicamentos = new ArrayList<>();
    }
    //agregar y mostrar medicamentos
    public void agregarMedicamento(Medicamento medicamento) {
        medicamentos.add(medicamento);
    }
    public void mostrarMedicamentos() {
        for (Medicamento med : medicamentos) {
            System.out.println(med);
        }
    }
    // medicamento por nombre
    public Medicamento buscarMedicamento(String nombre) {
        for (Medicamento med : medicamentos) {
            if (med.getNombre().equalsIgnoreCase(nombre)) {
                return med;
            }
        }
        return null;
    }
    public int getSucursal() { return sucursal; }
    public String getDireccion() { return direccion; }
    @Override
    public String toString() {
        return "Farmacia: " + nombreFarmacia + " | Sucursal: " + sucursal + " | Dirección: " + direccion;
    }
}

File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/per5python/farmacia.py
from medicamento import Medicamento

class Farmacia:
    def __init__(self, nombre_farmacia, sucursal, direccion):
        self.nombre_farmacia = nombre_farmacia
        self.sucursal = sucursal
        self.direccion = direccion
        self.medicamentos = [] 

    def __str__(self):
        return f"Farmacia: {self.nombre_farmacia} | Sucursal: {self.sucursal} | Dirección: {self.direccion}"

    def agregar_medicamento(self, medicamento):
        """Añadir un medicamento a la farmacia"""
        self.medicamentos.append(medicamento)

    def mostrar_medicamentos(self):
        """Mostrar todos los medicamentos disponibles"""
        for med in self.medicamentos:
            print(med)

    def busca_medicamento(self, nombre):
        """Buscar un medicamento por nombre"""
        for med in self.medicamentos:
            if med.nombre.lower() == nombre.lower():
                return med
        return None
